test/foundry/unit/concrete/erc4337account/TestERC4337Account_ValidateUserOp.t.sol

```diff
@@ -83,16 +83,19 @@ contract TestERC4337Account_ValidateUserOp is Test, NexusTest_Base {
     }
 
     /// @notice Tests user operation validation with an invalid nonce.
-    function test_ValidateUserOp_InvalidNonce() public {
+    function test_RevertWhen_InvalidNonce() public {
         PackedUserOperation[] memory userOps = new PackedUserOperation[](1);
-        uint256 incorrectNonce = 123; // deliberately incorrect
+        uint256 correctNonce = getNonce(address(BOB_ACCOUNT), MODE_VALIDATION, address(0));
+        uint256 incorrectNonce = correctNonce + 1; // deliberately incorrect to simulate invalid nonce
         userOps[0] = buildPackedUserOp(signer.addr, incorrectNonce);
         bytes32 userOpHash = ENTRYPOINT.getUserOpHash(userOps[0]);
         userOps[0].signature = signMessage(BOB, userOpHash);
 
+        // Expect the custom error InvalidModule to be thrown
+        vm.expectRevert(abi.encodeWithSelector(InvalidModule.selector, 0x0000000000000000000000000000000000000000));
```

Comment:
We should have more meaningful errors, like here we could have InvalidNonce instead of InvalidModule

Reply:
https://github.com/bcnmy/nexus/pull/121#discussion_r1713693858